### ISSUE TITLE
Incorrect unreachable code error for type alias after function return

### ISSUE THREAD

**Issue by danquirk** [palantir/tslint#564]
```ts
function foo() {
  return 1;
  function bar() {}
  type Bar = string;
}
```
Reports unreachable code on the type alias. 

Using tslint v2.5.0-dev.1
